src/docker-outside-of-docker-compose/README.md

```diff
@@ -47,6 +47,26 @@ Then reference the env var when running Docker commands from the terminal inside
 docker run -it --rm -v "${LOCAL_WORKSPACE_FOLDER//\\/\/}:/workspace" debian bash
 ```
 
+If anyway and like in Github Codespaces you want to be able to mount the local workspace folder into a container you create while running from inside the dev container **without** using [Docker in Docker](../docker-in-docker) to avoid a potential performance penalty you can use the **host**'s paths to mount the workspace folder **in the same place in the container that it is on the host,** so you can bind workspace contents as you would normally.
+
+Add the following to `.devcontainer.json`:
+
+```json
+"mounts": [
```

Comment:
I tested this with a dev container on a Mac OS. 
However, the container failed with this error 👇 

> docker: Error response from daemon: invalid mount config for type "bind": invalid mount path: '{localEnv:PWD}' mount path must be absolute.

I don't think `PWD` is available? Can you correct my understanding and help fix it for mac os?

```jsonc
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/docker-outside-of-docker
{
	"name": "Docker outside of Docker",
	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
	"image": "mcr.microsoft.com/devcontainers/base:bullseye",
	"features": {
		"ghcr.io/devcontainers/features/docker-outside-of-docker:1": {
			"version": "latest",
			"enableNonRootDocker": "true",
			"moby": "true"
		}
	},
	// Use this environment variable if you need to bind mount your local source code into a new container.
	"remoteEnv": {
		"LOCAL_WORKSPACE_FOLDER": "${localWorkspaceFolder}"
	},

	"mounts": [
		"source={localEnv:PWD},target={localEnv:PWD},type=bind,consistency=cached"
	],

	"workspaceFolder": "${localEnv:PWD}"
	// Use 'forwardPorts' to make a list of ports inside the container available locally.
	// "forwardPorts": [],
	// Use 'postCreateCommand' to run commands after the container is created.
	// "postCreateCommand": "docker --version",
	// Configure tool-specific properties.
	// "customizations": {},
	// Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
	// "remoteUser": "root"
}
```

Reply:
Also, how much of a performance difference are you seeing with this Template vs `docker-in-docker`? 

I strongly feel that one should use `docker-in-docker` when he'd like to do something of this sort as `docker-in-docker` is meant for it.

Reply:
@samruddhikhandale Sorry I missed 2 `$` arround `{localEnv:PWD}` inside the `mounts` array. 

Please try a new test with `"source=${localEnv:PWD},target=${localEnv:PWD},type=bind,consistency=cached"`.

Currently I use `docker-outside-docker` instead of `docker-in-docker` because I want to work on the sources currently on my disk when I am usind my PC or Mac.

I need to align my both workspaces paths on my docker host (desktop) and inside the dev container because my functional tests containers uses bind mount of my workspace.

I also does not like the idea of having to checkout the sources inside a `docker-in-docker` container volume because this is mandatory if it is mandatory to mount the project sources inside a test container that is run from inside the dev container.

I need also for some dev steps to update some generated file. The tools that are making the current update are run from inside an already builded docker image. This way we have not to configure the update tools on every developer desktop.

Also a penalty for me with `docker-in-docker` is that you cannot share the docker already pulled image, volumes with dev build caches, etc... accross different projects.